### PR TITLE
gen-docs.js ~ widget argument type support

### PR DESCRIPTION
**Pull request by undefinedDarkness**
Please follow this template, if applicable.

## Description
It looks for the types of args while generating docs

## Usage

When adding a widget or anything else that affects the configuration,
please provide a minimal example configuration snippet showcasing how to use it and

### Showcase

When adding widgets, please provide screenshots showcasing how your widget looks.
This is not strictly required, but strongly appreciated.

## Additional Notes
Sorry for garbage javascript ~ im not very well versed in anything newer than es6

## Checklist

Please make sure you can check all the boxes that apply to this PR.

- [ ] All widgets I've added are correctly documented.
- [ ] The documentation in the `docs/content/main` directory has been adjusted to reflect my changes.
- [ ] I used `cargo fmt` to automatically format all code before committing

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Why does this modify Cargo.lock?

**Comment by undefinedDarkness** [comment]
Because im shit at git and have no idea how to fix

**Comment by legendofmiracles** [comment]
When checked out on the can branch run this: `git rebase -i eb804ae3467200a7b0d162001fea02575805b2e7`

Then `drop` the first two commits (instead of pick).

**Comment by elkowar** [comment]
This currently crashes :/

image

**Comment by undefinedDarkness** [comment]
Is your node not up to date or smth?

**Comment by elkowar** [comment]
image
also this just means that propMatch[1] is not a string (i.e. undefined)

**Comment by undefinedDarkness** [comment]
Oh hmm ~ will look into it

**Comment by undefinedDarkness** [comment]
That ought to fix it

File: gen-docs.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        .join("\n\n");
+    console.log(output);
+}
+
+function printWidget(widget: Widget) {
+    return `
+## \`${widget.name}\` ${widget.desc ? `\n${widget.desc}` : ""}
+
+**Properties**
+${widget.props.map((prop) => `- [${prop.type}] **\`${prop.name}\`**: ${prop.desc}`).join("\n")}
```
Comment (elkowar) [inline gen-docs.ts:118]: I'd prefer a format like
```
` - **\`${prop.name}\`**: _\`${prop.type}\`_ ${prop.desc}`
```

Something like this:
image

**Comment by elkowar** [comment]
Also, I'd prefer if we turn `f64` and `i32` into `float` and `int` respectively in the docs, just replacing them in the types. f64 and i32 are kinda weird names for "user"-facing docs here.

**Comment by elkowar** [comment]
Thanks a ton, this is _much_ cleaner than it was before! Love deno!

**Pull request merged**